netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -79,14 +79,18 @@
    * noticing changes to DNS.
    */
   public static NettyChannelBuilder forAddress(SocketAddress serverAddress) {
```

Comment:
We _must not_ do this change. The resolved address may not have been created via DNS. In fact, during testing, that is quite common. If we are given a InetSocketAddress, there is no way we can safely re-resolve the IP.